src/tools/imageSplitter/src/imageSplitter.cpp

```diff
@@ -64,13 +77,16 @@ bool ImageSplitter::configure(yarp::os::ResourceFinder &rf)
         outRightPortName = rf.find("nameRight").asString();
     }
 
-    if(!rf.check("remote"))
+    yarp::os::ConstString remotePortName;
+
+    if(rf.check("remote"))
     {
-        yError() << "Missing 'remote' parameter. Please insert the name of the port to connect to.";
-        return false;
-    }
+        remotePortName = rf.find("remote").asString();
+        shouldConnect = true;
 
-    yarp::os::ConstString remotePortName = rf.find("remote").asString();
+    }
+    else
+        yWarning()<<"ImageSplitter: missing --remote parameter, external connection needed to " + inputPortName;
```

Comment:
This is supposed to be the normal behaviour when `--remote` is not passed, so this warninig looks a bit too much to me... Perhaps just add a

```c++
yInfo() << "ImageSplitter: waiting for connection to port" << inputPortName
```

Also this shouldn't be **here**, but after the port is actually open since it should print the **real** name of the port (i.e. the name of the port can change after `YARP_PORT_PREFIX` and `YARP_PORT_RENAME` are applied).